src/Raven.Client/ServerWide/Operations/CompareExchangeAsync.cs

```diff
@@ -17,32 +19,68 @@ public class RawClusterValueResult
         public long Index;
     }
 
-    public class ClusterValueResult<T>
+    public class CmpXchgResult<T>
     {
         public T Value;
         public long Index;
+
+
+        public static CmpXchgResult<T> ParseFromBlittable(BlittableJsonReaderObject response, DocumentConventions conventions)
+        {
+            if (response.TryGet(nameof(RawClusterValueResult.Index), out long index) == false)
+                throw new InvalidDataException("Response is invalid.");
+
+            response.TryGet(nameof(RawClusterValueResult.Value), out BlittableJsonReaderObject raw);
+
+            T result;
+            object val = null;
+            raw?.TryGet("Object", out val);
+
+            if (val == null)
+            {
+                return new CmpXchgResult<T>
+                {
+                    Index = index,
+                    Value = default(T)
+                };
+            }
+            if (val is BlittableJsonReaderObject obj)
+            {
+                result = (T)EntityToBlittable.ConvertToEntity(typeof(T), "cluster-value", obj, conventions);
+            }
+            else
+            {
+                raw.TryGet("Object", out result);
+            }
+
+            return new CmpXchgResult<T>
+            {
+                Index = index,
+                Value = result
+            };
+        }
     }
 
-    public class GetClusterValue<T> : IServerOperation<ClusterValueResult<T>>
+    public class GetCompareExchangeValueAsync<T> : IOperation<CmpXchgResult<T>>
     {
         private readonly string _key;
 
-        public GetClusterValue(string key)
+        public GetCompareExchangeValueAsync(string key)
         {
             _key = key;
         }
 
-        public RavenCommand<ClusterValueResult<T>> GetCommand(DocumentConventions conventions, JsonOperationContext context)
+        public RavenCommand<CmpXchgResult<T>> GetCommand(IDocumentStore store, DocumentConventions conventions, JsonOperationContext context, HttpCache cache)
         {
-            return new GetClusterValueCommand(_key, conventions);
+            return new GetCompareExchangeValueAsyncCommand(_key, conventions);
         }
 
-        private class GetClusterValueCommand : RavenCommand<ClusterValueResult<T>>
+        private class GetCompareExchangeValueAsyncCommand : RavenCommand<CmpXchgResult<T>>
```

Comment:
Rename to GetCompareExchangeValueCommand?

Reply:
Yes